File: leetcode/275. H 指数 II.java
class Solution {
    public int hIndex(int[] citations) {
        if (citations.length == 1) {
            return citations[0] > 0 ? 1 : 0;
        }
        for (int i = citations.length; i >= 1; i--) {
            if (citations[citations.length - i] >= i) {
                return i;
            }
        }
        return 0;
    }
}

File: leetcode/275. H 指数 II.rs
impl Solution {
    pub fn h_index(citations: Vec<i32>) -> i32 {
        if citations.len() == 1 {
            return if citations[0] > 0 { 1 } else { 0 };
        }
        for i in (1..=citations.len() as i32).rev() {
            if citations[citations.len() - i as usize] >= i {
                return i;
            }
        }
        return 0;
    }
}